src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState } from "react";
+import "./Modal.css";
+import ImageUploading, { ImageListType } from "react-images-uploading";
+import { Post } from "../App/App";
+// import { idText } from "typescript";
+
+interface ModalProps {
+  submitPost: (newPost: Post) => void;
+  closeModal: (event: any) => void;
+}
+
+export const Modal: React.FC<ModalProps> = ({ submitPost, closeModal }) => {
+  const [images, setImages] = useState([]);
+  const [postTitle, setTitle] = useState<string>("");
+  const [postDescription, setDescription] = useState<string>("");
+  const [imageURLS, setImageURLS] = useState<string[]>([]);
+
+  const maxNumber = 2;
+  const onChange = (
+    imageList: ImageListType,
+    addUpdateIndex: number[] | undefined
+  ) => {
+    setImages(imageList as never[]);
+    let onlyURLs = imageList.map((image) => image.data_url);
+    setImageURLS(onlyURLs);
+  };
+
+  const clearState = () => {
+    setImages([]);
+    setTitle("");
+    setDescription("");
+  };
+
+  const addPost = (event: React.MouseEvent<HTMLElement>) => {
+    event.preventDefault();
+    const newPost = {
+      id: 2,
+      title: postTitle,
+      description: postDescription,
+      image: imageURLS,
+      user: { id: 4, username: "workerBee1" },
+      upvotes: 0,
+      downvotes: 0,
+      created_at: "August 29, 2021",
+    };
+    submitPost(newPost);
+    clearState();
+  };
+
+  return (
+    <section
+      className="modal-wrapper"
+      onClick={(event) => closeModal(event)}
+      id="modalWrapper"
+    >
+      <section className="modal-content" id="modalContent">
+        <i className="fas fa-times"></i>
+        <label>Create Post</label>
```

Comment:
@k-atwhite There seems to be an "X" shown at the top of the modal above "Create Post", but when I was messing with this, it doesn't seem like I can use that to close the modal (?) The modal opens and closes fine, but only closes by clicking outside it.

Reply:
Yeah before merging can you make is so that the X closes it as well?